src/shims/tsd.h

```diff
@@ -73,6 +73,56 @@ _dispatch_thread_key_create(const unsigned long *k, void (*d)(void *))
 	if (!*k || !d) return;
 	dispatch_assert_zero(pthread_key_init_np((int)*k, d));
 }
+#elif DISPATCH_USE_THREAD_LOCAL_STORAGE
+
+DISPATCH_TSD_INLINE
+static inline void
+_dispatch_thread_key_create(pthread_key_t *k, void (*d)(void *))
+{
+	dispatch_assert_zero(pthread_key_create(k, d));
+}
+
+struct dispatch_tsd {
+    bool initialized;
+    pid_t tid;
+    void* dispatch_queue_key;
+    void* dispatch_voucher_key;
+#if DISPATCH_USE_OS_SEMAPHORE_CACHE
+#error "Invalid DISPATCH_USE_OS_SEMAPHORE_CACHE configuration"
+#else
+    void* dispatch_sema4_key;
+#endif
+    void* dispatch_cache_key;
+    void* dispatch_io_key;
+    void* dispatch_apply_key;
+    void* dispatch_defaultpriority_key;
+#if DISPATCH_INTROSPECTION
+    void* dispatch_introspection_key;
+#elif DISPATCH_PERF_MON
+    void* dispatch_bcounter_key;
+#endif
+    void* dispatch_pthread_root_queue_observer_hooks_key;
+};
+
+extern __thread struct dispatch_tsd __dispatch_tsd;
+extern pthread_key_t __dispatch_tsd_key;
+extern void libdispatch_thread_init(void);
+
+DISPATCH_ALWAYS_INLINE
+static inline
+struct dispatch_tsd *
+_dispatch_get_tsd_base(void)
+{
+    if (slowpath(!__dispatch_tsd.initialized))
```

Comment:
please add braces.

`if (a) some_short_stmt;` is sometimes ok if `a` is short, but if you use multiline, always use braces
and here, multiline is the right way